### ISSUE TITLE
resnet2_AdaFM.py

### ISSUE THREAD

**Issue by slala2121** [MiaoyunZhao/GANTransferLimitedData#1]
Should the input into the forward method for the generator be yz? i.e.,

style_w=self.small_style(yz)
out=self.small_fc(yz)

**Comment by MiaoyunZhao** [comment]
> Should the input into the forward method for the generator be yz? i.e.,
> 
> style_w=self.small_style(yz)
> out=self.small_fc(yz)

It depends on whether the label information is considered. If the label information is considered, a conditional GAN model is used with yz as the input; when no label information is considered, which is the case in this paper, the input will be z.

**Comment by slala2121** [comment]
Thanks for your quick response. 
When I tried incorporating this change, the code throws an error on dimensionality compatibility issues in the forward pass. Do I need to change the "8" in the line
out = out.view(batch_size, 8 * self.small_nf, self.s0, self.s0)? Where does the 8 come from?

**Comment by MiaoyunZhao** [comment]
> Thanks for your quick response. 
> When I tried incorporating this change, the code throws an error on dimensionality compatibility issues in the forward pass. Do I need to change the "8" in the line
> out = out.view(batch_size, 8 * self.small_nf, self.s0, self.s0)? Where does the 8 come from?

This is a dimensional inconsistent problem. You should make the following changes (in "resnet2_AdaFM.py") when label information is considered:
line 386: self.small_fc = nn.Linear(z_dim, 8 * small_nf * s0 * s0)
--> self.small_fc = nn.Linear(z_dim+embed_size, 8 * small_nf * s0 * s0)
line 420: layers.append(EqualLinear(z_dim, style_dim))
--> layers.append(EqualLinear(z_dim+embed_size, style_dim))
line 439: style_w = self.small_style(z)
--> style_w = self.small_style(yz)
line 441: out = self.small_fc(z)
--> out = self.small_fc(yz)
Uncomment line 543~546
Also make changes in the config file (e.g. Flowers.yaml) accordingly:
line 12 and 18: embed_size: 1
--> embed_size: 256

By the way, 8 is an experimental setting for the capacity/wide of the generator.

**Comment by slala2121** [comment]
Thanks. Why do we need to increase the embedding size?

**Comment by MiaoyunZhao** [comment]
> Thanks. Why do we need to increase the embedding size?

This is a typical setting in many existing works, e.g. GP-GAN. Actually, there are also many other ways to consider the label information (e.g. BigGAN), you can try and find the one fit your problem best.

**Comment by slala2121** [comment]
Ok thanks!
